### PR TITLE
MIDI module and docs

### PR DESCRIPTION
**Pull request by DonutCables**
Adds the `midi.py` module assembled by xs5871 on Discord/Matrix and includes the `modules.md` and `midi.md` documentation for implementing it.

### CODE REVIEW

**Comment by kdb424** [comment]
I'll let @xs5871 review this one as it's based on their code.

**Comment by kdb424** [comment]
Merging is also blocked due to formatting. We really should get around to a contribution "getting started" sort of page that tells you to run `make fix-isort fix-formatting test` before committing.

File: kmk/modules/midi.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Originally put together by xs5871 on the KMK Firmware Discord/Matrix channel
```
Comment (xs5871) [inline kmk/modules/midi.py:1]: Thanks for mention, but that's not necessary. It's not like we're linking to another resource for reference.

File: docs/midi.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+## Keycodes
+
+|Key         |Description                                                                  |
+|-----------------|------------------------------------------------------------------------|
+|`KC.MIDI_CC()`      |Sends a ControlChange message; accepts two integer arguments of `0`-`15`(controller number) then `0`-`127`(control value)                                           |
+|`KC.MIDI_NOTE()`      |Sends a Note message with both 'On' and 'Off' segments; accepts two integer arguments of `0`-`127`(note number) and `0`-`127`(velocity)            |
+|`KC.MIDI_PB()` |Sends a Pitch Wheel message; accepts a single integer argument of `0`-`16383`, centered on `8192`                                 |
+|`KC.MIDI_PC()`  |Sends a Program Change message; accepts a single integer argument of `0`-`127`(program number)               |
+|`KC.MIDI_START()`      |Sends a Start message; accepts no arguments          |
+|`KC.MIDI_STOP()`      |Sends a Stop message; accepts no arguments                     |
```
Comment (xs5871) [inline docs/midi.md:18]: * Add arguments to the key codes, like: `KC.MIDI_NOTE(note, velo)`
* table formating (raw text) is a bit off
* MIDI channel selection is missing

I don't think you have to describe the MIDI protocol in detail in the key code table, I'm sure everyone who needs MIDI will be familiar with value ranges and that they have to be integers.

Comment (DonutCables) [reply docs/midi.md:18]: - Will be added, forgot to go back to it
 - Yeah idk how I let it get that bad
 - Is that a command or a parameter? I was just going off the keys laid out in `midi.py`

Comment (xs5871) [reply docs/midi.md:18]: The key codes have the same signature as the adafruit_midi messages: `KC.MIDI_NOTE(note, velocity, channel=0)`, where `channel` is a named optional argument. No need to list that in the table, just mention it in a paragraph below.

**Comment by xs5871** [comment]
It looks like you merged in master instead of rebasing your PR. Don't know if that'll cause issues.

**Review by xs5871 — APPROVED** [review]

**Comment by kdb424** [comment]
It should auto resolve AFIK

**Review by kdb424 — APPROVED** [review]

**Pull request merged**